contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,218 @@
+use bcs::to_bytes;
+use cosmwasm_std::{HexBinary, Uint256};
+use itertools::Itertools;
+
+use crate::error::ContractError;
+
+use super::Data;
+
+// TODO: all of the public functions in this file should be moved to a trait,
+// that has an abi and bcs implementation (and possibly others)
+
+pub fn command_params(
+    source_chain: String,
+    source_address: String,
+    destination_address: String,
+    payload_hash: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    if payload_hash.len() != 32 {
+        return Err(ContractError::InvalidMessage {
+            reason: format!("payload hash is not 32 bytes {}", payload_hash.to_hex()),
+        });
+    }
+
+    let destination_address = <[u8; 32]>::try_from(
+        HexBinary::from_hex(&destination_address)?.to_vec(),
+    )
+    .map_err(|_| ContractError::InvalidMessage {
+        reason: format!(
+            "destination_address is not a valid Sui address: {}",
+            destination_address
+        ),
+    })?;
```

Comment:
I think it's me to blame, but what is `ret` lol?

Reply:
lol you hate my short variable names, it's supposed to stand for "return". I changed it though